util/reggen/field.py

```diff
@@ -140,9 +141,10 @@ def from_raw(reg_name: str,
                              'hardware write')
 
         bits = Bits.from_raw(where, reg_width, params, rd['bits'])
+        if (bits.bitmask() & used_bitmask != 0):
```

Comment:
I'd suggest:
```python
       if bits.bitmask() & used_bitmask:
```
You don't need the parens around the outside in Python and the `!=` operator means you need to think about precedence between that and `&`. Typing less seems to actually be clearer in this case! :-)

However, I think this check should probably be moved into `Register`: see below.